### PR TITLE
High perf bar series

### PR DESCRIPTION
**Pull request by jacobp100**
Pending https://github.com/ag-grid/ag-charts/pull/2998

### CODE REVIEW

File: packages/ag-charts-community/src/chart/data/dataController.ts [hunk 1]
```diff
@@ -179,11 +179,14 @@ export class DataController {
...
+
         return ([group]: RequestedProcessing<any, any, any>[]) =>
-            (opts.groupByData === false || group.data === data) &&
-            group.opts.groupByKeys === opts.groupByKeys &&
-            group.opts.groupByFn === opts.groupByFn &&
-            keys(group.opts.props) === keys(opts.props);
+            (groupByData === false || group.data === data) &&
+            (group.opts.groupByKeys ?? false) === groupByKeys &&
+            group.opts.groupByFn === groupByFn &&
+            keys(group.opts.props) === propsKeys;
```
Comment (jacobp100) [inline packages/ag-charts-community/src/chart/data/dataController.ts:189]: Ensures financial charts requests are merged - needed so the domains are referentially equal for all series

File: packages/ag-charts-community/src/chart/series/cartesian/barSeries.ts [hunk 2]
```diff
@@ -369,7 +407,7 @@ export class BarSeries extends AbstractBarSeries<Rect, BarSeriesProperties, BarN
                 midPoint: { x: rect.x + rect.width / 2, y: rect.y + rect.height / 2 },
                 fill,
                 stroke,
-                opacity: 1,
+                opacity,
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/series/cartesian/barSeries.ts:410]: Does this impact series highlighting? It maybe that opacity was being applied at the series level, which is why it wasn't set here?

File: packages/ag-charts-types/src/chart/navigatorOptions.ts [hunk 3]
```diff
@@ -146,7 +146,7 @@ const __MINI_CHART_SERIES_OPTIONS = undefined as any as Record<
     NonNullable<AgMiniChartSeriesOptions['type']> | IgnoredMiniChartSeries,
     string
 >;
-// @ts-expect-error TS6133 - this is used to validate completeness by the compiler, but is deliberately unused.
+// @ts-ignore TS6133 - this is used to validate completeness by the compiler, but is deliberately unused.
```
Comment (alantreadway) [inline packages/ag-charts-types/src/chart/navigatorOptions.ts]: Revert, should be fixed on latest now.

File: packages/ag-charts-types/src/chart/themeOptions.ts [hunk 4]
```diff
@@ -325,6 +325,6 @@ type VerifyAgBaseChartThemeOverrides<T = AgBaseChartOptions> = {
 
 // Verification checks for completeness/correctness.
 const __THEME_OVERRIDES = undefined as any as Required<AgChartThemeOverrides>;
-// @ts-expect-error TS6133 - this is used to validate completeness by the compiler, but is deliberately unused.
+// @ts-ignore TS6133 - this is used to validate completeness by the compiler, but is deliberately unused.
```
Comment (alantreadway) [inline packages/ag-charts-types/src/chart/themeOptions.ts]: Ditto

File: packages/ag-charts-types/src/chartBuilderOptions.ts [hunk 5]
```diff
@@ -164,7 +164,7 @@ export interface AgChartInstance<O extends AgChartInstanceOptions = AgChartOptio
     destroy(): void;
 }
 
-// @ts-expect-error Expected to be unused in code, simplified typings to enable rendering for API docs.
+// @ts-ignore - Expected to be unused in code, simplified typings to enable rendering for API docs.
```
Comment (alantreadway) [inline packages/ag-charts-types/src/chartBuilderOptions.ts]: Ditto

File: packages/ag-charts-types/src/presets/gauge/commonOptions.ts [hunk 6]
```diff
@@ -32,7 +32,7 @@ export interface AgBaseGaugeThemeableOptions {
 
 // Verification checks for completeness/correctness.
 const __THEMEABLE_OPTIONS = undefined as any as Required<AgBaseGaugeThemeableOptions>;
-// @ts-expect-error TS6133 - this is used to validate completeness by the compiler, but is deliberately unused.
+// @ts-ignore TS6133 - this is used to validate completeness by the compiler, but is deliberately unused.
```
Comment (alantreadway) [inline packages/ag-charts-types/src/presets/gauge/commonOptions.ts]: Ditto

File: packages/ag-charts-types/src/presets/gauge/commonOptions.ts [hunk 7]
```diff
@@ -65,7 +65,7 @@ export interface AgGaugeScaleLabel {
 
 // Verification checks for completeness/correctness.
 const __AXIS_LABEL_OPTIONS = undefined as any as Required<AgGaugeScaleLabel>;
-// @ts-expect-error TS6133 - this is used to validate completeness by the compiler, but is deliberately unused.
+// @ts-ignore TS6133 - this is used to validate completeness by the compiler, but is deliberately unused.
```
Comment (alantreadway) [inline packages/ag-charts-types/src/presets/gauge/commonOptions.ts]: Ditto

File: packages/ag-charts-community/src/chart/series/cartesian/barSeries.ts (hunk 2 above)
Comment (jacobp100) [reply packages/ag-charts-community/src/chart/series/cartesian/barSeries.ts:410]: Nope. createNodeData isn't called for series highlighting

**Review by alantreadway — APPROVED** [review]
LGTM

**Pull request merged**